vault/logical_system.go

```diff
@@ -692,6 +693,42 @@ func NewSystemBackend(core *Core, config *logical.BackendConfig) (logical.Backen
 				HelpSynopsis:    strings.TrimSpace(sysHelp["audited-headers"][0]),
 				HelpDescription: strings.TrimSpace(sysHelp["audited-headers"][1]),
 			},
+			&framework.Path{
+				Pattern: "plugin-catalog/$",
+
+				Fields: map[string]*framework.FieldSchema{},
+
+				Callbacks: map[logical.Operation]framework.OperationFunc{
+					logical.ListOperation: b.handlePluginCatalogList,
+				},
+
+				HelpSynopsis:    strings.TrimSpace(sysHelp["plugin-catalog"][0]),
+				HelpDescription: strings.TrimSpace(sysHelp["plugin-catalog"][1]),
+			},
+			&framework.Path{
+				Pattern: "plugin-catalog/(?P<name>.+)",
+
+				Fields: map[string]*framework.FieldSchema{
+					"name": &framework.FieldSchema{
+						Type: framework.TypeString,
+					},
+					"sha_256": &framework.FieldSchema{
+						Type: framework.TypeString,
+					},
+					"command": &framework.FieldSchema{
+						Type: framework.TypeString,
```

Comment:
I can update the description here, but it's actually the former. e.g. `command="my-plugin --server"` would work. The catalog then looks in the plugin directory for a file named `my-plugin` and adds any subsequent parameters to a slice.